tests/phpunit/src/Commands/CopyCronTasksCommandTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Acquia\Cli\Tests\Commands;
+
+use Acquia\Cli\Command\CopyCronTasksCommand;
+use Acquia\Cli\Tests\CommandTestBase;
+use Symfony\Component\Console\Command\Command;
+
+/**
+ * Class CopyCronTasksCommandTest.
+ *
+ * @property \Acquia\Cli\Command\CopyCronTasksCommand $command
+ * @package Acquia\Cli\Tests\Commands
+ */
+class CopyCronTasksCommandTest extends CommandTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function createCommand(): Command {
+    return $this->injectCommand(CopyCronTasksCommand::class);
+  }
+
+  /**
+   * @group rohit
+   * Tests the 'app:cron-copy' command.
+   *
+   * @throws \Exception
+   */
+  public function testCopyCronTasksCommandTest(): void {
+    $this->executeCommand([
+        'source_app' => 'app.test',
+        'dest_app' => 'app.test'
+      ],
+    );
+    $output = $this->getDisplay();
+    $this->assertStringContainsString('The source and destination environments can not be same', $output);
```

Comment:
You need to mock the calls to Cloud API.

Something like...

```
 /**
   * @group rohit
   * Tests the 'app:cron-copy' command.
   *
   * @throws \Exception
   */
  public function testCopyCronTasksCommandTest(): void {
    $environments_response = $this->getMockEnvironmentsResponse();
    $this->mockForGetEnvironmentFromAliasArg();
    $source_crons_list_response = $this->getMockResponseFromSpec('/environments/{environmentId}/crons', 'get', '200');
    $this->clientProphecy->request('get',
      '/environments/' . $environments_response->{'_embedded'}->items[0]->uuid) . '/crons'
      ->willReturn($source_crons_list_response)
      ->shouldBeCalled();
    // @todo Mock create cron jobs.
    $this->executeCommand([
        'source_app' => 'app.prod',
        'dest_app' => 'app.test'
      ],
    );
    $output = $this->getDisplay();

}
```

Reply:
I checked this on local by mocking the response but problem I am seeing is that in the all available mock responses, we don't have two different app alias (checked in acquia-spec.yml file) and thus this is failing in very early.

Reply:
You can change the response after you load it from the spec--it's just a php variable.

Reply:
Finally able to run test successfully on local but noticed some issues in mock response and how handle by the upstream.

Example-1 - 
I got error - 
```
array_map(): Expected parameter 2 to be an array, object given
vendor/typhonius/acquia-php-sdk-v2/src/Response/CronsResponse.php:19
```
This is because the mock api response is object but `CronsResponse` expects this to be array as call array_map( and thus issue on local.


Example-2
AcquiaCloudApi\Endpoints\create() mock and response returned by $this->client->request('post', "/environments/${environmentUuid}/crons", $options) doesn't match

as response returned by api not have `message` key and thus OperationResponse failed as this has 
```
$this->message = $operation->message;
```

Reply:
And now We see same are failing in CI jobs as well.

Reply:
I can try to debug it if you'd like. Seems like you've gotten close to completing things.

Reply:
@grasmash i will need ur help 
(if you have sometime) for this failure as not sure even after mock api returning object but the method is returning null as checked in error trace and thus the issue